hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/ImmediateExecutorService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.fixtures;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An {@link ExecutorService} that executes tasks immediately in the calling thread, rather than in a separate thread.
+ * This is useful for synchronous testing where the API involves an {@link ExecutorService}.
```

Comment:
The requirement of everything to be synchronized may slow down testing quite substantially (not sure).  Based on how the AbstractExecutorService (which has no synchronized methods) calls execute inline rather than managing any queue itself, I believe we could implement this correctly without synchronized by declaring shutdown to be volatile.
The only point of potential issue I see would be if we have calls to InvokeAll, which _might_ still complete one or more tasks from that set on its calling thread after another thread called shutdown.
In a testing context, I don't believe that would be significant (though we should verify that).

Reply:
So far I haven't seen any slowdown in the tests. You may be right that we don't need to use synchronized, but doing so makes it obviously correct, whereas without it, I would need to devote more brainpower and validation to it to make sure it is correct. At this point, I don't think the payoff is worth it. But we could improve this in a follow on PR if it is a problem (but I really don't think it will be, since synchronization is really very fast in wall clock terms, it is only a problem under contention, and in this case we should almost never have contention since most tests are single threaded).